src/components/CSnackbar.vue

```diff
@@ -0,0 +1,120 @@
+<!--
+Copyright (c) 2024 by Crystal Creations GbR and Johannes Huther
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+associated documentation files (the "Software"), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute,
+sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
+THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ -->
+<template>
+  <div>
+    <v-snackbar
+      :model-value="modelValue"
+      :timeout="snackbarTimeout"
+      :location="position"
+      :color="type"
+      @update:model-value="onModelValueChange"
+    >
+      <slot></slot>
+    </v-snackbar>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from "vue";
+
+/**
+ * Available types for a snackbar.
+ *
+ * Either one of the defined types or a theme color.
+ */
+export type CSnackbarType = "error" | "success" | string;
+
+/**
+ * Available positions for a snackbar.
+ */
+export type CSnackbarPosition = "top" | "bottom";
+
+const props = withDefaults(
+  defineProps<{
+    /**
+     * The model, whether the snackbar should be shown or not.
+     */
+    modelValue: boolean;
+
+    /**
+     * The {@link CSnackbarType} of this snackbar.
+     *
+     * The {@link CSnackbarType} changes the color and the timeout of this snackbar.
+     */
+    type?: CSnackbarType;
+
+    /**
+     * The {@link CSnackbarPosition} of this snackbar.
+     */
+    position?: CSnackbarPosition;
+
+    /**
+     * Timeout in milliseconds of this snackbar after which it will be hidden.
+     *
+     * Set to -1 or 0 to display forever.
+     */
+    timeout?: number | null;
+  }>(),
+  {
+    type: "error",
+    position: "bottom",
+    timeout: null,
+  },
+);
+
+const emit = defineEmits<{
+  /**
+   * Event emitted when the component's model changes.
+   */
+  (e: "update:modelValue", showSnackbar: boolean): void;
+
+  /**
+   * Event emitted when the snackbar is closed.
+   */
+  (e: "closed"): void;
+}>();
+
+/**
+ * Returns the timeout to use for this snackbar.
+ *
+ * If a custom {@link timeout} is given, this will be used.
+ *
+ * Otherwise, the timeout is the default or depends on the {@link type}:
+ * - error: 5000ms
+ * - default: 3000ms
+ */
+const snackbarTimeout = computed<number>(() => {
```

Comment:
it is, see docs for the timeout property. I checked the vuetify docs again and changed the docs a bit to fit the newest version (https://vuetifyjs.com/en/api/v-snackbar/#props-timeout).

I also added `infinite` as a type to the timeout property to make it easier to use since I think, -1 is not the most memorable.